### ISSUE TITLE
problem with synchronize

### ISSUE THREAD

**Issue by ShadowBelmolve** [wikimatze/hammer.vim#1]
Error detected while processing function PreviewMKD:
line   90:
NoMethodError: undefined method `synchronize' for #Mutex:0x9b8c4b0
E488: Trailing characters
Press ENTER or type command to continue

ArchLinux
VIM - Vi IMproved 7.2 (2008 Aug 9, compiled May 16 2010 23:11:27)
ruby 1.9.1p378 (2010-01-10 revision 26273) [i686-linux]

**Comment by ghost** [comment]
Hi,  
I've come across this bug on Arch Linux too, but I do not think it is a bug related to vim-markdown-preview. The bug seems to be related to ruby support for vim on Arch Linux.

You should be able to reproduce the bug by sending vim the command:  
`:ruby require('rubygems')`

I haven't reported the bug to the Arch Linux maintainer(s) - I do not have an account yet.
I will leave this ticket open until the issue is resolved.

Thanks,  
Rob

**Comment by ghost** [comment]
Just an update, I have reported this bug to the Arch Linux bug tracker.  
You can follow it [here](http://bugs.archlinux.org/task/19954)

**Comment by ShadowBelmolve** [comment]
I've opened a bug report, here http://bugs.archlinux.org/task/19955

**Comment by ghost** [comment]
I have found a workaround for this bug.  
I will push it upstream soon.

**Comment by ghost** [comment]
Unfortunately, my work around has hit another bug in ruby support for vim on Arch.  
I do not know why support for ruby in vim is so broken on Arch. The first bug has been assigned, and hopefully the second one will be assigned shortly. I will keep working to see if i can beat the bug without writing too much boiler plate code.

(See http://bugs.archlinux.org/task/19991 for details on the new bug)

**Comment by ghost** [comment]
Hi,  
I managed to get Arch Linux generating previews without a problem, and simplified the install process. 

See the `arch` branch:
http://github.com/robgleeson/vim-markdown-preview/tree/arch 

I plan to merge this into master sometime soon. Please let me know if it works for you!  
I will close this issue when you can confirm it is working for you.

**Comment by ghost** [comment]
I'm closing this issue.  
I merged the work done on the `arch` branch into `master`, and it is working for me on OSX and Arch Linux. Maybe when the Arch issues with vim are resolved, we will use system() inside of VIM.command(':! ..') but we'll see how it goes.  

Thanks, 
Rob

**Comment by artm** [comment]
I'm seeing the same error with vim on windows: "undefined method synchronize" whether issuing `:ruby require('rubygems')` or at hammer initialization. That's vim 7.3 and ruby 1.9.1 (the same as vim was compiled against).

**Comment by ghost** [comment]
Hi artm,  

I think this issue has to do with _Vim_ being built against _Ruby 1.9_.
I can only recommend the advice given on the _Arch Linux_ forums, which suggests upgrading _Vim_ to >= 7.3.88

**Comment by artm** [comment]
I've figured out a couple of things:
- Official windows build of vim (that I use) supports dynamic loading of ruby (has `+ruby/dyn` in the output of `:version`). 
- I had ruby 1.9.1 installed and `c:\Ruby191\bin` in `$PATH` - that was the one used by vim.
- This problem, whatever it is, seems to be [1.9.1 specific](http://vim.1045645.n5.nabble.com/gvim-editing-ruby-on-Windows-td3243792.html)
- I have installed Ruby 1.9.2 and changed the `$PATH` to contain `c:\Ruby192\bin` instead.

Now this problem went away, but now the next line hit me (#9).
